pkg/generators/openapi.go

```diff
@@ -662,3 +674,95 @@ func (g openAPITypeWriter) generateSliceProperty(t *types.Type) error {
 	g.Do("},\n},\n},\n", nil)
 	return nil
 }
+
+// CustomArgs is used by the gengo framework to pass args specific to the API linter
```

Comment:
Thoughts on the added lines from here on out: Can we put this in its own package, maybe more than one? I think a package that collects the violation definition & file reading/writing makes sense? At the very least, this belongs in a new file.

Reply:
I thought `kube-openapi/pkg/validator` sounds more like an OpenAPI spec validator not an API rule validator. These lines are specific to `openAPIGen` but not any other generator under `kube-openapi/pkg/generators`, so I put them here. I can create a new file and we can think about organizing in some package.